packages/edit-site/src/components/editor/index.js

```diff
@@ -126,10 +117,10 @@ export default function EditSiteEditor( { isPostsList = false } ) {
 			hasSiteIcon: !! siteData?.site_icon_url,
 		};
 	}, [] );
-	const postWithTemplate = !! contextPostId;
+	const postWithTemplate = !! context?.postId;
 	useEditorTitle(
-		postWithTemplate ? contextPostType : editedPostType,
-		postWithTemplate ? contextPostId : editedPostId
+		postWithTemplate ? context.postId : postId,
+		postWithTemplate ? context.postType : postType
```

Comment:
The order here needs to be swapped - `useEditorTitle( postType, postId )`.

Currently, theeditor renders `undefined` in the title`.